### PR TITLE
Fixed alignment errors in several subsystems

### PR DESCRIPTION
**Pull request by tridge**
Fixes warnings in:
 - ExpandingArray
 - HAL_Linux
 - RCProtocol SRXL2
 - AP_Generator RicenPower

Ping @peterbarker  @andyp1per @rmackay9

### CODE REVIEW

File: libraries/AP_Common/AP_ExpandingArray.h [hunk 1]
```diff
@@ -95,15 +95,23 @@ class AP_ExpandingArray : public AP_ExpandingArrayGeneric
     T &operator[](uint16_t i)
     {
         const uint16_t chunk_num = i / chunk_size;
-        const uint16_t chunk_index = (i % chunk_size) * elem_size;
-        return (T &)(chunk_ptrs[chunk_num][chunk_index]);
+        const uint16_t chunk_index = (i % chunk_size);
+        #pragma GCC diagnostic push
```
Comment (rmackay9) [inline libraries/AP_Common/AP_ExpandingArray.h:99]: nitpick: slightly indenting issue with the top pragma

File: libraries/AP_Common/AP_ExpandingArray.h [hunk 2]
```diff
@@ -95,15 +95,23 @@ class AP_ExpandingArray : public AP_ExpandingArrayGeneric
...
     }
 
     // allow use as an array for accessing elements. no bounds checking is performed
     const T &operator[](uint16_t i) const
     {
         const uint16_t chunk_num = i / chunk_size;
-        const uint16_t chunk_index = (i % chunk_size) * elem_size;
-        return (const T &)(chunk_ptrs[chunk_num][chunk_index]);
+        const uint16_t chunk_index = (i % chunk_size);
+        #pragma GCC diagnostic push
```
Comment (rmackay9) [inline libraries/AP_Common/AP_ExpandingArray.h:111]: small indenting issue here too..

**Comment by rmackay9** [comment]
I guess this should have no functional impact?  It just resolves warnings?

**Comment by tridge** [comment]
> I guess this should have no functional impact? It just resolves warnings?

On some MCUs it could prevent a fault and crash, if the MCU doesn't support misaligned accesses

File: libraries/AP_Common/AP_ExpandingArray.h (hunk 1 above)
Comment (tridge) [reply libraries/AP_Common/AP_ExpandingArray.h:99]: fixed

**Comment by tridge** [comment]
> Does this clean all of the instances of this up?

it cleared up all the ones in the navio build. The remaining warnings are the string related ones

**Comment by andyp1per** [comment]
LGTM

**Comment by tridge** [comment]
> AutoTests are failing as `be16toh` has been changed to to `le16h_ptr` in the generator code. That took me a while to work out...

fixed now

**Pull request merged**